### ISSUE TITLE
djangorestframework_camel_case.camelize_serializer_fields does not handle "ignore_keys"

### ISSUE THREAD

**Issue by tomashchuk** [tfranzel/drf-spectacular#945]
**drf_spectacular.contrib.djangorestframework_camel_case.camelize_serializer_fields** does not handle "ignore_keys"

Define for example    
```
REST_FRAMEWORK = {
    "JSON_UNDERSCOREIZE": {
          "ignore_keys": ("version_8test", ),
    },
} 
```   
then add

```
SPECTACULAR_SETTINGS = {
    'POSTPROCESSING_HOOKS': [
        'drf_spectacular.contrib.djangorestframework_camel_case.camelize_serializer_fields'
    ],
}
```

**Actual behavior**
`{"version8test": 4294967295}`

**Expected behavior**
`{"version_8test": 4294967295}`

Let me know if it is expected behavior by default otherwise, I will create a PR with the fix.

**Comment by tfranzel** [comment]
good catch! looks like a bug ant not expected behavior. I think this was kind of working before, but broke with a bugfixing another issue.

we used `camelize` (which includes the ignore list) before, but that was only intended for modifying data and not a schema as far as I remember. there were issues like #861. we went one step deeper and used their building blocks instead. it was an incomplete transition from the looks of it.

yes, PR is welcome.

code is here:

https://github.com/tfranzel/drf-spectacular/blob/master/drf_spectacular/contrib/djangorestframework_camel_case.py

**Comment by tomashchuk** [comment]
@tfranzel It could be a possible solution for the issue with "ignore_keys". But [djangorestframework_camel_case](https://github.com/vbabiy/djangorestframework-camel-case#ignore-fields) also handles "ignore_fields" and it needs to be solved too. I could not run tests locally because of  `django.core.exceptions.ImproperlyConfigured: Could not find the GDAL library`. Maybe I will try later to setup it up locally and fix these issues but I did not run tests for these changes. Tested only with my project.

```
import re


def camelize_serializer_fields(result, generator, request, public):
    from djangorestframework_camel_case.util import camelize_re, underscore_to_camel
    from djangorestframework_camel_case.settings import api_settings

    # ignore_fields = api_settings.JSON_UNDERSCOREIZE.get("ignore_fields") or () should be handled too
    ignore_keys = api_settings.JSON_UNDERSCOREIZE.get("ignore_keys") or ()

    def camelize_str(key: str):
        new_key = re.sub(camelize_re, underscore_to_camel, key) if "_" in key else key
        if key in ignore_keys or new_key in ignore_keys:
            return key
        return new_key

    def camelize_component(schema: dict):
        if schema.get('type') == 'object':
            if 'properties' in schema:
                schema['properties'] = {
                    camelize_str(field_name): camelize_component(field_schema)
                    for field_name, field_schema in schema['properties'].items()
                }
            if 'required' in schema:
                schema['required'] = [camelize_str(field) for field in schema['required']]
        elif schema.get('type') == 'array':
            camelize_component(schema['items'])
        return schema

    for (_, component_type), component in generator.registry._components.items():
        if component_type == 'schemas':
            camelize_component(component.schema)

    # inplace modification of components also affect result dict, so regeneration is not necessary
    return res
```

**Comment by tfranzel** [comment]
sry about that GDAL thing. I need to find a mitigation for that. Can't expect anyone to install `libgdal` for this.

uncomment this line and the tests should run (exept of course the 2 GIS tests):

https://github.com/tfranzel/drf-spectacular/blob/6300b1258b008b557da879ac5c58cb0b2c97fa4d/tests/conftest.py#L23

**Comment by tfranzel** [comment]
`./runtests.py` should now work without the library installed.  full `tox` run will require it though.

**Comment by tomashchuk** [comment]
@tfranzel Ok, I will try when having some time

**Comment by tfranzel** [comment]
closed with release `0.26.0`
